wrapper/src/main/java/software/amazon/jdbc/Driver.java

```diff
@@ -96,10 +96,21 @@ public Connection connect(final String url, final Properties info) throws SQLExc
     LOGGER.finest("Opening connection to " + url);
 
     final String driverUrl = url.replaceFirst(PROTOCOL_PREFIX, "jdbc:");
-    final java.sql.Driver driver = DriverManager.getDriver(driverUrl);
+
+    final String message = Messages.get(
+        "Driver.missingDriver",
+        new Object[] {driverUrl, Collections.list(DriverManager.getDrivers())});
+
+    java.sql.Driver driver;
+    try {
+      driver = DriverManager.getDriver(driverUrl);
+    } catch (SQLException e) {
+      LOGGER.warning(() -> message);
+      throw new SQLException(message, e);
+    }
 
     if (driver == null) {
-      LOGGER.warning(() -> Messages.get("Driver.missingDriver", new Object[] {driverUrl}));
+      LOGGER.warning(() -> message);
```

Comment:
1: is it needed to log it here, since it was logged inside the catch block?

2: since it is a failure, why is it logger.warning instead of logger.error ?